core/assert.c

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 2016 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author  Martine Lenders <mlenders@inf.fu-berlin.de>
+ */
+
+#include "assert.h"
+
+#ifdef DEBUG_ASSERT_VERBOSE
+NORETURN void _assert_failure(const _assert_failure_info_t *info)
+{
+    printf("%s:%d => ", info->file, info->line); \
```

Comment:
I'm not sure this would make things smaller BTW. Currently `assert_crash_message` and the file names are stored at distinct locations. Building a new message for the `core_panic` message parameter would create a new string for each file name + `assert_crash_message` (duplicating that and `assert_crash_message`'s data).